server/controllers/auth.js

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable consistent-return */
+import jwt from 'jsonwebtoken';
+
+import config from '../config';
+import User from '../models/User';
+
+function invalidResponse(response, message) {
+  return response.status(400).json({
+    success: false,
+    message,
+    user: null,
+  });
+}
+
+function successfulResponse(response, message, userData, token) {
+  return response.status(200).json({
+    success: true,
+    message,
+    user: { ...userData, token },
+  });
+}
+
+function getToken(userData) {
+  return jwt.sign(userData, config.JWT_SECRET_KEY, { expiresIn: '7d' });
+}
+
+function passwordPatternCheck(password) {
+  const pattern = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*_=+-]).{8,}$');
+  return pattern.test(password);
+}
+
+function getUserData(user) {
+  const {
+    id,
+    email,
+    profileImageUrl,
+    gender,
+    DOB,
+    name,
+  } = user;
+  const userData = {
+    id,
+    email,
+    profileImageUrl,
+    gender,
+    DOB,
+    name,
+  };
+  return userData;
+}
+
+exports.signIn = async function (request, response) {
+  // finding the user
+  // password matches?
+  // log them in: sending jwt
+  try {
+    const user = await User.findOne({
+      email: request.body.email,
+    });
+    const userData = getUserData(user);
+    const isMatch = await user.comparePassword(request.body.password);
+    if (isMatch) {
+      const token = getToken(userData);
+      return successfulResponse(response, 'Sign-in successful', userData, token);
+    }
+    return invalidResponse(response, 'Invalid Email/Password.');
+  } catch (error) {
+    return invalidResponse(response, 'Invalid Email/Password.');
```

Comment:
Here error is different. Why are you sendinf invalid email/pwd?

Reply:
Modified.